linode/manager
# test: [M3-7891] - Improve Cypress feature flag mocking ergonomics

## Description 📝
This is a backwards compatible change to how we mock LaunchDarkly feature flags from our Cypress tests. The goal of this change is to make mocking feature flags less burdensome when writing Cypress tests.

Previously, mocking LaunchDarkly features required using the `mockAppendFeatureFlags`, `mockGetFeatureFlagClientstream`, and `makeFeatureFlagData` utils. With this change in place, only `mockAppendFeatureFlags` is required to mock feature flags in most cases. This is a backwards compatible change, however, so existing calls to `mockGetFeatureFlagClientstream`, `mockAppendFeatureFlags`, and `makeFeatureFlagData` will continue to work as expected.

This PR also modifies a few tests for the sake of demonstrating the change, but the remaining clean up of our feature flag mocks will be handled by M3-8321.

## Changes  🔄
- Call `mockGetFeatureFlagClientstream` before every test, making it unnecessary to call from within tests
- Modify `mockAppendFeatureFlags` to accept a `FeatureFlagMockData` object, which is a new type that allows feature flags to be expressed more flexibly
- Clean up feature flag mocks in `betas.spec.ts`, `create-image.spec.ts`, and `placement-groups-navigation.spec.ts` to demonstrate the change
- Removes `mockGetFeatureFlags` since we've never used it and don't have a use case for it

## How to test 🧪
We can rely on CI to confirm that these changes work. Specifically, we want to ensure that existing calls to `mockAppendFeatureFlags`, etc., continue to work as expected (this is the majority of our tests), and that the modified tests (`betas.spec.ts`, `create-image.spec.ts`, and `placement-groups-navigation.spec.ts`) pass too.

## As an Author I have considered 🤔

*Check all that apply*
